### PR TITLE
Make Exclude local networks a configurable setting

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205852060630307/f

### Description
See asana

### Steps to test this PR

_Test exclude local routes_
- [x] install from this branch and enable NetP
- [x] go to netp management screen -> VPN settings
- [x] toggle "Exclude local networks" on
- [x] navigate back
- [x] verify vpn restarts
- [ ] verify in logcat (filter by `Adding route`) that proper routes are added

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3786** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3786" target="_blank">Graphite</a> 
    * **PR #3787** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3787" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3787?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/NetPSettingsLocalConfig.kt [hunk 1]
```diff
@@ -46,6 +46,12 @@ interface NetPSettingsLocalConfig {
 
     @Toggle.DefaultValue(true)
     fun vpnNotificationAlerts(): Toggle
+
+    /**
+     * When `true` the VPN routes will exclude local networks
+     */
+    @Toggle.DefaultValue(false)
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/NetPSettingsLocalConfig.kt:53]: Yes, you're right. I got mixed up with the exclusion

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/WgVpnRoutes.kt [hunk 2]
```diff
@@ -33,7 +33,7 @@ internal class WgVpnRoutes {
      *   - broadcast address
      */
     companion object {
-        val wgVpnRoutes: Map<String, Int> = mapOf(
+        val wgVpnDefaultRoutes: Map<String, Int> = mapOf(
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/WgVpnRoutes.kt:36]: Yes, you're right. I got mixed up with the exclusion

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/NetPDefaultConfigProvider.kt [hunk 3]
```diff
@@ -48,8 +43,28 @@ data class PcapConfig(val filename: String, val snapLen: Int, val fileSize: Int)
...
     private val netPExclusionListRepository: NetPExclusionListRepository,
+    private val dispatcherProvider: DispatcherProvider,
+    private val netPSettingsLocalConfig: NetPSettingsLocalConfig,
 ) : NetPDefaultConfigProvider {
     override fun exclusionList(): Set<String> {
         return netPExclusionListRepository.getExcludedAppPackages().toSet()
     }
+
+    override suspend fun routes(): Map<String, Int> = withContext(dispatcherProvider.io()) {
+        val defaultImpl = object : NetPDefaultConfigProvider {}
```
Comment (karlenDimla) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/NetPDefaultConfigProvider.kt:54]: NIT: this is not used anymore

**Review by karlenDimla — APPROVED** [review]

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/NetPDefaultConfigProvider.kt (hunk 3 above)
Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/NetPDefaultConfigProvider.kt:54]: damn, I was took quick merging, we'll remove it in another PR. Sorry about that

**Pull request merged**